test/functional/url-expander/test-expander.js

```diff
@@ -0,0 +1,870 @@
+      const res = expander
+          .expand('?a=CLIENT_ID(AMP_ECID_GOOGLE,,_ga)&b=CLIENT_ID(_ga)');
+      return expect(res).to.eventually.match(/^\?a=12345.54321&b=12345.54321/);
+    });
+
+    it('should replace VARIANT', () => {
+      sandbox.stub(Services, 'variantForOrNull')
+          .returns(Promise.resolve({
+            x1: 'v1',
+            x2: 'none',
+          }));
+      const url = '?x1=VARIANT(x1)&x2=VARIANT(x2)&x3=VARIANT(x3)';
+      return expect(expander.expand(url))
+          .to.eventually.equal('?x1=v1&x2=none&x3=');
+    });
+
+    it('should replace VARIANT with empty string if ' +
+        'amp-experiment is not configured ', () => {
+      const url = '?x1=VARIANT(x1)&x2=VARIANT(x2)&x3=VARIANT(x3)';
+      return expect(expander.expand(url))
+          .to.eventually.equal('?x1=&x2=&x3=');
+    });
+
+    it('should replace VARIANTS', () => {
+      sandbox.stub(Services, 'variantForOrNull')
+          .returns(Promise.resolve({
+            x1: 'v1',
+            x2: 'none',
+          }));
+      return expect(expander.expand('?VARIANTS')).to
+          .eventually.equal('?x1.v1!x2.none');
+    });
+
+    it('should replace VARIANTS with empty string if ' +
+        'amp-experiment is not configured ', () => {
+      return expect(expander.expand('?VARIANTS')).to.eventually.equal('?');
+    });
+
+    it('should replace SHARE_TRACKING_INCOMING and' +
+        ' SHARE_TRACKING_OUTGOING', () => {
+      const fragments = Promise.resolve({
+        incomingFragment: '12345',
+        outgoingFragment: '54321',
+      });
+      sandbox.stub(Services, 'shareTrackingForOrNull')
+          .returns(fragments);
+      const url = '?in=SHARE_TRACKING_INCOMING&out=SHARE_TRACKING_OUTGOING';
+      const res = expander.expand(url);
+      return expect(res).to.eventually.equal('?in=12345&out=54321');
+    });
+
+    it('should replace SHARE_TRACKING_INCOMING and SHARE_TRACKING_OUTGOING' +
+        ' with empty string if amp-share-tracking is not configured', () => {
+      const url = '?in=SHARE_TRACKING_INCOMING&out=SHARE_TRACKING_OUTGOING';
+      return expect(expander.expand(url))
+          .to.eventually.equal('?in=&out=');
+    });
+
+    it('should replace STORY_PAGE_INDEX and STORY_PAGE_ID', () => {
+      const stories = Promise.resolve({
+        pageIndex: 546,
+        pageId: 'id-123',
+      });
+      sandbox.stub(Services, 'storyVariableServiceForOrNull')
+          .returns(stories);
+      const res = expander.expand('?index=STORY_PAGE_INDEX&id=STORY_PAGE_ID');
+      return expect(res).to.eventually.equal('?index=546&id=id-123');
+    });
+
+    it('should replace STORY_PAGE_INDEX and STORY_PAGE_ID' +
+        ' with empty string if amp-story is not configured', () => {
+      return expect(
+          expander.expand('?index=STORY_PAGE_INDEX&id=STORY_PAGE_ID'))
+          .to.eventually.equal('?index=&id=');
+    });
+
+    it('should replace TIMESTAMP', () => {
```

Comment:
Maybe I didn't make it clear. Those variables are not implemented in this expander, hence the tests should not go here.

They should be in `url-replacement-impl-test.js`. If we start to extract GlobalVariableSource to a separate file, then we can consider have them in `test-global-variable-source.js` at that time.

Reply:
removed most of the tests